.github/workflows/adhoc_fuzzer.yml

```
@@ -0,0 +1,245 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: "Adhoc Fuzzer Jobs"
+
+on:
+  workflow_dispatch:
+    branchRef:
+      description: 'Branch to checkout out'
+      default: 'main'
+    numThreads:
+      description: 'Number of threads'
+      default: 16
+    maxHighMemJobs:
+      description: 'Number of high memory jobs'
+      default: 8
+    maxLinkJobs:
+      description: 'Maximum number of link jobs'
+      default: 4
+    extraCMakeFlags:
+      description: 'Additional CMake flags'
+      default: '-DVELOX_ENABLE_ARROW=ON'
+
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  linux-presto-fuzzer-run:
+    runs-on: 16-core
+    timeout-minutes: 120
+    env:
+      CCACHE_DIR: "${{ github.workspace }}/.ccache/"
+      CCACHE_BASEDIR: "${{ github.workspace }}"
+      LINUX_DISTRO: "ubuntu"
+    steps:
+
+      - name: "Checkout Repo"
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+
+      - name: "Install dependencies"
+        run: source ./scripts/setup-ubuntu.sh
+
+      - name: "Build"
+        run: |
+          make debug NUM_THREADS="${{ github.event.inputs.numThreads }}" MAX_HIGH_MEM_JOBS="${{ github.event.inputs.maxHighMemJobs }}" MAX_LINK_JOBS="${{ github.event.inputs.maxLinkJobs }}" EXTRA_CMAKE_FLAGS="${{ github.event.inputs.extraCMakeFlags }}"
```

You can use the `inputs` context directly. See https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context for details
```suggestion
          make debug NUM_THREADS="${{ inputs.numThreads }}" MAX_HIGH_MEM_JOBS="${{ inputs.maxHighMemJobs }}" MAX_LINK_JOBS="${{ inputs.maxLinkJobs }}" EXTRA_CMAKE_FLAGS="${{ inputs.extraCMakeFlags }}"
```